File: python/simple-cipher/simple_cipher.py
from string import ascii_lowercase
from random import choice
class Cipher:
    def __init__(self, key=None):
        if not key:
            key = ''
            for i in range(100):
                key += choice(ascii_lowercase)
        self.key = key
        self.key_list = list(key)

    def encode(self, text):
        chars = list(text)
        out = []
        for i, char in enumerate(chars):
            offset = ord(self.key_list[i % len(self.key_list)]) - ord('a')
            out.append(chr((ord(char) - 97 + offset) % 26 + 97))
        return ''.join(out)

    def decode(self, text):
        chars = list(text)
        out = []
        for i, char in enumerate(chars):
            offset = ord(self.key_list[i % len(self.key_list)]) - ord('a')
            out.append(chr((ord(char) - 97 - offset) % 26 + 97))
        return ''.join(out)

File: ruby/simple-cipher/simple_cipher.rb
class Cipher

  attr_reader :key
  def rand_cipher
    100.times.map { ('a'..'z').to_a[Random.rand(26)] }.join
  end
  def initialize(key=rand_cipher)
    raise ArgumentError unless key.match(/^[a-z]+$/)
    @key = key
    @alpha = ('a'..'z').to_a
  end

  def encode(plaintext)
    @key.chars.cycle.first(plaintext.length).zip(plaintext.chars).map { |(cipher, char)|
      @alpha[(@alpha.index(cipher) + @alpha.index(char)) % 26]
    }.join
  end

  def decode(encodedtext)
    @key.chars.cycle.first(encodedtext.length).zip(encodedtext.chars).map { |(cipher, char)|
      @alpha[(@alpha.index(char) - @alpha.index(cipher)) % 26]
    }.join

  end

end
